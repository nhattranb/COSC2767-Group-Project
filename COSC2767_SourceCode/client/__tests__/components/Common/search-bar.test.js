/*
 *   RMIT University Vietnam
 *   Course: COSC2767 Systems Deployment and Operations
 *   Semester: 2024C
 *   Assessment: Assignment 2
 *   Author: Le Dinh Ngoc Quynh
 *   ID: s3791159
 *   Created  date: 17 Dec 2024
 *   Last modified: 17 Dec 2024
 */


import React from 'react';
import {fireEvent, render, waitFor} from '@testing-library/react';
import SearchBar from "../../../app/components/Common/SearchBar";
import {expect, testm, jest, test} from "@jest/globals";

test('renders SearchBar correctly', () => {
    const { asFragment } = render(<SearchBar />);
    expect(asFragment()).toMatchSnapshot();
});

test('calls onBlur prop when input loses focus', () => {
    const onBlur = jest.fn();
    const { getByPlaceholderText } = render(<SearchBar onBlur={onBlur} />);
    const input = getByPlaceholderText('Search');

    fireEvent.blur(input, { target: { value: 'test' } });

    expect(onBlur).toHaveBeenCalledWith({ name: 'search', value: 'test' });
})

test('calls onSearch prop when typing', async () => {
    const onSearch = jest.fn();
    const { getByPlaceholderText } = render(<SearchBar onSearch={onSearch} />);
    const input = getByPlaceholderText('Search');

    fireEvent.change(input, { target: { value: 'test' } });

    await waitFor(() => {
        expect(onSearch).toHaveBeenCalledWith({ name: 'search', value: 'test' });
    }, { timeout: 1500 });
});


test('calls onBlur prop when input loses focus', () => {
    const onBlur = jest.fn();
    const { getByPlaceholderText } = render(<SearchBar onBlur={onBlur} />);
    const input = getByPlaceholderText('Search');

    fireEvent.blur(input, { target: { value: 'test' } });

    expect(onBlur).toHaveBeenCalledWith({ name: 'search', value: 'test' });
})
