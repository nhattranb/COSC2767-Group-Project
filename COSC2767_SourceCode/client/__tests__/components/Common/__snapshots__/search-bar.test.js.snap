// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders SearchBar correctly 1`] = `
<DocumentFragment>
  <form
    novalidate=""
  >
    <div
      class="search-box inline-btn-box"
    >
      <div
        class="input-text-block"
      >
        <input
          autocomplete="off"
          class="input-text search-box"
          id="search"
          name="search"
          placeholder="Search"
          type="text"
          value=""
        />
        <button
          class="input-btn custom-btn-primary md text-only icon-left "
          style="border-radius: 3px;"
          type="submit"
        >
          <span
            class="btn-text"
          >
            Search
          </span>
        </button>
      </div>
    </div>
  </form>
</DocumentFragment>
`;
